src/ui/settings/components/InstallPluginAlert.tsx

```diff
@@ -0,0 +1,64 @@
+import { clipboard } from "@metro/common";
+import { installPlugin } from "@lib/plugins";
+import { findByProps } from "@metro/filters";
+import { Forms } from "@ui/components";
+
+const { FormInput } = Forms;
+const Alert = findByProps("alertDarkStyles", "alertLightStyles").default;
+
+const Alerts = findByProps("openLazy", "close");
+
+export function showInstallPluginAlert() {
+	Alerts.openLazy({
+		importer: async function () {
+			return () => <InstallPluginAlert />;
+		}
+	});
+};
+
+const HTTP_REGEX = /^https?:\/\/(?:www\.)?[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,6}\b(?:[-a-zA-Z0-9()@:%_\+.~#?&\/=]*)$/;
+
+export default function InstallPluginAlert() {
+	const [pluginUrl, setPluginUrl] = React.useState("");
+	const [error, setError] = React.useState("");
+
+	React.useEffect(() => {
+		clipboard.getString().then((c) => { if (HTTP_REGEX.test(c)) setPluginUrl(c) });
+	}, []);
+
+	const installPluginCallback = () => {
+		installPlugin(pluginUrl).then(() => {
+			setPluginUrl("");
+			Alerts.close();
+		}).catch((e: Error) => {
+			setError(e.message);
+		});
+	};
+
```

Comment:
I'd prefer if this callback was directly embedded into the component hierarchy like your other callbacks are.

Reply:
This gets called by onConfirm from the Alert and onSubmitEditing from the FormInput. Having this shared between the two like this reduces duplication.

Reply:
Oops, didn't see that. In which case, please move it to a `function` instead of an arrow.